src/components/AppController.tsx

```diff
@@ -119,8 +68,8 @@ const updateQueryHashParams = (hashParams: UrlHashParams): boolean => {
  */
 const handleHashChange = (ev: Nullable<HashChangeEvent>): UrlHashParams => {
```

Comment:
If the individual update helpers will modify the URL parameters, we should be returning the parsed hash parameters before the helpers are called, right?

Can we make `handleHashChange` return void (nothing) instead? And replace the "Handle initial page load and maintain full URL state" logic as:
```
        // Handle initial page load and maintain full URL state
        handleHashChange(null);
        const hashParams = getWindowUrlHashParams();
        const searchParams = getWindowUrlSearchParams();
```